app/jsx/pages/SearchPage.jsx

```diff
@@ -343,12 +343,12 @@ class FacetFieldset extends React.Component {
     let out = []
     for (let facet of facets) {
       if (facet.value in checked)
-        out.push(facet)
-      else if (out.length < 5)
+        out.unshift(facet)
```

Comment:
I know this is not alphabetic. That's a whole other issue I am working on. And will touch this up once that is fixed.

Reply:
This is an improvement. However, it's not quite right in this situation:
1. Start with no filters, expand Department facet, then Show More
2. Scroll down and select a child without its parent, e.g. Center for the Build Environment.
3. Close the Show More box, and the selection is not visible (because I didn't select the parent).

This will be tricky to fix. I hope you're up for it, because I dread trying to figure it out.

All that said, what you've done here is an improvement so I'm planning to merge and push this as-is for now.